Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/Visuals/ProximityEffect/ProximityEffect.cs

```diff
@@ -137,30 +137,34 @@ public void UpdateScaling(Vector3 boundsCenter, Vector3 boundsExtents)
             {
                 foreach (var pointer in inputSource.Pointers)
                 {
-                    if (pointer.IsInteractionEnabled && !proximityPointers.Contains(pointer))
+                    // don't use IsInteractionEnabled for near pointers as the pointers might have a different radius when deciding
+                    // if they can interact with a near-by object - we might still want to show proximity scaling even if
+                    // eg. grab pointer decides it's too far away to actually perform the interaction
+                    if (pointer.IsActive && !proximityPointers.Contains(pointer))
                     {
                         proximityPointers.Add(pointer);
                     }
                 }
             }
 
-            // Get the max radius possible of our current bounds plus the proximity
-            float maxRadius = Mathf.Max(Mathf.Max(boundsExtents.x, boundsExtents.y), boundsExtents.z);
-            maxRadius *= maxRadius;
-            maxRadius += config.ObjectCloseProximity + config.ObjectMediumProximity;
+            // Get the max radius possible of our current bounds and extent the range to include proximity scaled objects
+            float squareMaxLength = boundsExtents.sqrMagnitude + (3 * config.ObjectMediumProximity * config.ObjectMediumProximity);
```

Comment:
Where does this 3 come from? Is it that `(3 * config.ObjectMediumProximity * config.ObjectMediumProximity)` is equivalent to the square magnitude of a vector where xyz are all `config.ObjectMediumProximity`?

Reply:
yes exactly that! I'm expanding the existing bounds by the medium proximity value. i'll add a comment